objectherkenning_openbare_ruimte/databricks_pipelines/post_processing_pipeline/data_enrichment_step/run_data_enrichment_step.py

```diff
@@ -0,0 +1,230 @@
+    get_job_process_time,
+)
+from objectherkenning_openbare_ruimte.databricks_pipelines.common.table_manager import (  # noqa: E402
+    TableManager,
+)
+from objectherkenning_openbare_ruimte.databricks_pipelines.post_processing_pipeline.data_enrichment_step.components import (  # noqa: E402
+    utils_visualization,
+)
+from objectherkenning_openbare_ruimte.databricks_pipelines.post_processing_pipeline.data_enrichment_step.components.clustering_detections import (  # noqa: E402
+    Clustering,
+)
+from objectherkenning_openbare_ruimte.databricks_pipelines.post_processing_pipeline.data_enrichment_step.components.decos_data_connector import (  # noqa: E402
+    DecosDataHandler,
+)
+from objectherkenning_openbare_ruimte.databricks_pipelines.post_processing_pipeline.data_enrichment_step.components.vulnerable_bridges_handler import (  # noqa: E402
+    VulnerableBridgesHandler,
+)
+from objectherkenning_openbare_ruimte.settings.databricks_jobs_settings import (  # noqa: E402
+    load_settings,
+)
+
+
+def run_data_enrichment_step(
+    sparkSession,
+    catalog,
+    schema,
+    root_source,
+    vuln_bridges_relative_path,
+    az_tenant_id,
+    db_host,
+    db_name,
+    job_process_time,
+):
+    # Setup clustering
+    clustering = Clustering(spark=sparkSession, catalog=catalog, schema=schema)
+    clustering.setup()
+    containers_coordinates_geometry = clustering.get_containers_coordinates_geometry()
+
+    # Setup bridges data
+    bridgesHandler = VulnerableBridgesHandler(
+        spark=sparkSession,
+        root_source=root_source,
+        vuln_bridges_relative_path=vuln_bridges_relative_path,
+    )
+    bridges_coordinates_geometry = bridgesHandler.get_bridges_coordinates_geometry()
+
+    # Setup permit data
+    decosDataHandler = DecosDataHandler(
+        spark=sparkSession,
+        az_tenant_id=az_tenant_id,
+        db_host=db_host,
+        db_name=db_name,
+        db_port=5432,
+    )
+
+    tableManager = TableManager(spark=sparkSession, catalog=catalog, schema=schema)
+
+    print(f"03: Number of containers: {len(containers_coordinates_geometry)}.")
+    # Enrich with bridges data
+    (
+        closest_bridges_distances,
+        closest_bridges_ids,
+        closest_bridges_coordinates,
+        closest_bridges_wkts,
+    ) = VulnerableBridgesHandler.calculate_distances_to_closest_vulnerable_bridges(
+        bridges_locations_as_linestrings=bridges_coordinates_geometry,
+        containers_locations_as_points=containers_coordinates_geometry,
+        bridges_ids=bridgesHandler.get_bridges_ids(),
+        bridges_coordinates=bridgesHandler.get_bridges_coordinates(),
+    )
+
+    clustering.add_columns(
+        {
+            "closest_bridge_distance": closest_bridges_distances,
+            "closest_bridge_id": closest_bridges_ids,
+            "closest_bridge_coordinates": closest_bridges_coordinates,
+            "closest_bridge_linestring_wkt": closest_bridges_wkts,
+        }
+    )
+
```

Comment:
use join instead of add_column.

pass detection_id everywhere and return dataframes instead of tuple of lists